factom/client.py

```diff
@@ -151,128 +126,86 @@ def current_minute(self):
         roundtimeout: returns the number of seconds between rounds of an
         election during a fault.
         """
-        return self._request('current-minute')
+        return self._request("current-minute")
 
-    def directory_block_by_height(self, height):
-        """
-        Retrieve a directory block given only its height.
-        """
-        return self._request('dblock-by-height', {
-            'height': height
-        })
+    def directory_block_by_height(self, height: int):
+        """Retrieve a directory block given only its height."""
+        return self._request("dblock-by-height", {"height": height})
 
-    def directory_block_by_keymr(self, keymr):
+    def directory_block_by_keymr(self, keymr: Union[bytes, str]):
         """
-        Every directory block has a KeyMR (Key Merkle Root), which can
-        be used to retrieve it. The response will contain information
-        that can be used to navigate through all transactions (entry
-        and factoid) within that block. The header of the directory
-        block will contain information regarding the previous directory
-        block keyMR, directory block height, and the timestamp.
+        Every directory block has a KeyMR (Key Merkle Root), which can be used to retrieve it.
+        The response will contain information that can be used to navigate through all transactions (entry
+        and factoid) within that block. The header of the directory block will contain information regarding
+        the previous directory block key merkle root, directory block height, and the timestamp.
         """
-        return self._request('directory-block', {
-            'keymr': keymr
-        })
+        return self._request("directory-block", {"keymr": keymr if type(keymr) is str else keymr.hex()})
 
     def directory_block_head(self):
         """
-        The directory block head is the last known directory block by
-        factom, or in other words, the most recently recorded block.
-        This can be used to grab the latest block and the information
+        The directory block head is the last known directory block by factom, or in other words,
+        the most recently recorded block. This can be used to grab the latest block and the information
         required to traverse the entire blockchain.
         """
-        return self._request('directory-block-head')
+        return self._request("directory-block-head")
 
-    def entry(self, hash):
+    def entry(self, hash: Union[bytes, str], encode_as_hex: bool = False):
```

Comment:
Might be a good idea to use a different identifier for the variable instead of `hash`, as hash is also a reserved keyword.

Reply:
Yep good point, might as well change those names now while we're updating the functions. Done